pkg/tfbridge/detailed_diff_test.go

```diff
@@ -1747,3 +1752,573 @@ func TestDetailedDiffPulumiSchemaOverride(t *testing.T) {
+					propertyMapElems("val1", "val2", "val3"),
+					propertyMapElems("val3", "val2", "val1", "val1"), tfs, ps, map[string]*pulumirpc.PropertyDiff{})
+			})
+
+			t.Run("shuffled added front", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems("val2", "val3"),
+					propertyMapElems("val1", "val3", "val2"), tfs, ps, map[string]*pulumirpc.PropertyDiff{
+						"foo[0]": {Kind: add},
+					},
+				)
+			})
+
+			t.Run("shuffled added middle", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems("val1", "val3"),
+					propertyMapElems("val3", "val2", "val1"), tfs, ps, map[string]*pulumirpc.PropertyDiff{
+						"foo[1]": {Kind: add},
+					},
+				)
+			})
+
+			t.Run("shuffled added end", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems("val1", "val2"),
+					propertyMapElems("val2", "val1", "val3"), tfs, ps, map[string]*pulumirpc.PropertyDiff{
+						"foo[2]": {Kind: add},
+					},
+				)
+			})
+
+			t.Run("shuffled removed front", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems("val1", "val2", "val3"),
+					propertyMapElems("val3", "val2"), tfs, ps, map[string]*pulumirpc.PropertyDiff{
+						"foo[0]": {Kind: delete},
+					},
+				)
+			})
+
+			t.Run("shuffled removed middle", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems("val1", "val2", "val3"),
+					propertyMapElems("val3", "val1"), tfs, ps, map[string]*pulumirpc.PropertyDiff{
+						"foo[1]": {Kind: delete},
+					},
+				)
+			})
+
+			t.Run("shuffled removed end", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems("val1", "val2", "val3"),
+					propertyMapElems("val2", "val1"), tfs, ps, map[string]*pulumirpc.PropertyDiff{
+						"foo[2]": {Kind: delete},
+					},
+				)
+			})
+
+			t.Run("computed", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems("val1"),
+					resource.NewPropertyMapFromMap(
+						map[string]interface{}{
+							"foo": computedValue,
+						},
+					),
+					tfs, ps,
+					map[string]*pulumirpc.PropertyDiff{
+						"foo": {Kind: update},
+					},
+				)
+			})
+
+			t.Run("nil to computed", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					resource.NewPropertyMapFromMap(
+						map[string]interface{}{},
+					),
+					resource.NewPropertyMapFromMap(
+						map[string]interface{}{
+							"foo": computedValue,
+						},
+					),
+					tfs, ps,
+					map[string]*pulumirpc.PropertyDiff{
+						"foo": {Kind: pulumirpc.PropertyDiff_ADD},
+					},
+				)
+			})
+
+			t.Run("empty to computed", func(t *testing.T) {
+				runDetailedDiffTest(t,
+					propertyMapElems(),
+					resource.NewPropertyMapFromMap(
+						map[string]interface{}{
+							"foo": computedValue,
+						},
+					),
+					tfs, ps,
+					map[string]*pulumirpc.PropertyDiff{
+						"foo": {Kind: pulumirpc.PropertyDiff_UPDATE},
+					},
+				)
+			})
+		})
+	}
+}
+
+func TestDetailedDiffSetBlockNestedMaxItemsOne(t *testing.T) {
```

Comment:
At some point perhaps we'll have enough confidence in path-based methods not to worry about this one.